traffic_ops/app/lib/UI/DeliveryService.pm

```diff
@@ -921,6 +919,10 @@ sub update {
 			}
 		}
 
+		my $new_hostname = UI::SslKeys::get_hostname($self, $id, $update);
+		$upd_ssl = 1 if $old_hostname ne $new_hostname;
```

Comment:
This is good, but shouldn't the API also perform the same check and update SSL keys if the hostname changed?